src/core/search/search_parser_test.cc

```diff
@@ -51,6 +52,46 @@ class SearchParserTest : public ::testing::Test {
   QueryDriver query_driver_;
 };
 
+class MockedSearchInput : public SearchInput {
+ public:
+  using Map = std::unordered_map<std::string, std::string>;
+
+  MockedSearchInput() = default;
+  MockedSearchInput(std::string test_field) : hset_{{"field", test_field}} {
+  }
+
+  MockedSearchInput& operator=(Map hset) {
```

Comment:
I tend to dislike implicit-looking conversions such as this. It's also against the Google style guide.
Can you please use a traditional method, like `SetFrom()` or whatever?

Reply:
Ok, I'll do.

I just always thought the rules in the style guide are there for a reason. This one mainly to not have ambiguous or unexpected conversions/behaviour/etc with re-useable core components, as its easy to miss for others. Its clear where the danger is for large code bases.

I don't see any way to misuse it inside tests 😅 I just want it to be readable and contain as less visual clutter (brackets) as possible.